libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/BooleanExpression.kt

```diff
@@ -12,20 +14,25 @@ import kotlin.reflect.KClass
  * other that may also contain possibleTypes
  */
 sealed class BooleanExpression<out T : Any> {
-  /**
-   * This is not super well defined but works well enough for our simple use cases
-   */
+  @Deprecated(message = "Should not be used anymore", level = DeprecationLevel.ERROR)
```

Comment:
Maybe something more actionable, just in case?

```suggestion
  @Deprecated(message = "This was only used in internal API and shouldn't have been part of the public API. If you needed this, please open an issue.", level = DeprecationLevel.ERROR)
```